### ISSUE TITLE
Train using librispeech

### ISSUE THREAD

**Issue by kdavis-mozilla** [mozilla/DeepSpeech#333]
For issue #330 we want to benchamark (time per epoch, rate of WER decrease...),  DeepSpeech 2 vs our system.

DeepSpeech 2 uses as its training data librispeech. 

Thus to compare the two we need to train our system on librispeech. The goal of this issue is to do just that.

**Comment by kdavis-mozilla** [comment]
I've trained on librispeech.

The validation WER for all validation data sets is 22% for the clean validation data set it's 12%.

**Comment by baljeet** [comment]
@kdavis-mozilla 
Is there a different hardware requirement for training librispeech?
I have been trying to train it on p2.xlarge with 11.5 G VRAM on aws, getting resource exhausted.
Would appreciate if you can share your hardware config...

**Comment by kdavis-mozilla** [comment]
@baljeet We trained on a box with 4 Pascal Titan X GPU's each with 12GB of RAM and 64GB of host RAM. Each epoch took about 10h55m.

The hyperparameters are described in [run-librivox.sh](https://github.com/mozilla/DeepSpeech/blob/issue333/bin/run-librivox.sh). However,  they have yet to be ported to the corresponding file [run-librivox.sh](https://github.com/mozilla/DeepSpeech/blob/master/bin/run-librivox.sh) after the refactor. [PS: If you could do so the PR would be more than welcome :-)]

We did a display step and validation step every 5 epochs. However, one should note that early on, epoch 5 and 10 say, the WER report takes a long time to compute as the network then is making lots of spelling errors and the WER calculation is done in the CPU, which is slow. For example I think at epoch 5 the WER report took as long as one epoch. So you may want to do the display step and validation step less often.

**Comment by baljeet** [comment]
Did you say it took ~11 hours for one epoch?
How much was the GPU utilization during the process?
I was training using batchsize 32 and 50 epochs... Should i just reduce them to 15?

Also, it would be great if the reasoning behind all the hyperparameters is available somewhere...

**Comment by kdavis-mozilla** [comment]
@baljeet First, thanks for the PR! There are some little comments I made on it.

Yeah one epoch takes about 11h. GPU utilization is about 99% for the entire time.

From a few light tests we've done we've found that the 8 GPU instances on EC2 are about half as fast as our in-house server. So, we guess that the 16 GPU instances are about on parity with our in-house server. So, I'd guess the 16 GPU EC2 instances would take about 11h per epoch. However, I don't know what the batch sizes should be on the EC2 instances.

The logic behind the hyperparameters is mostly hard-won trial an error. The most expensive parameter to tune was the dropout rate which is much higher than that used in the original Deep Speech paper. So we had to do a binary search using the TED data set which took lots of compute time.
